main/springboot/src/main/java/org/mobilitydata/gtfsvalidator/springboot/GtfsValidatorController.java

```diff
@@ -0,0 +1,129 @@
+package org.mobilitydata.gtfsvalidator.springboot;
+
+import com.beust.jcommander.JCommander;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.BucketInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageClass;
+import com.google.cloud.storage.StorageOptions;
+import com.google.common.flogger.FluentLogger;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import org.mobilitydata.gtfsvalidator.cli.Arguments;
+import org.mobilitydata.gtfsvalidator.cli.Main;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+/**
+ * Holds the responsability to run the validation through a {@code SpringBoot} interface. Provides
+ * methods to push the validation report to a google cloud storage.
+ */
+public class GtfsValidatorController {
+
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+  private static final Gson DEFAULT_GSON =
+      new GsonBuilder()
+          .serializeNulls()
+          .setPrettyPrinting()
+          .serializeSpecialFloatingPointValues()
+          .create();
+  private static final String VALIDATION_REPORT_BUCKET = "VALIDATION_REPORT_BUCKET";
+  private static final String DEFAULT_OUTPUT_BASE = "output";
+  private static final String DEFAULT_NUM_THREADS = "8";
+  private static final String DEFAULT_COUNTRY_CODE = "null";
+
+  @GetMapping("/")
+  @ResponseBody
+  /**
+   * Runs {@code org.mobilitydata.gtfsvalidator.cli.Main.main} with the arguments extracted via the
+   * query parameters. Returns the status of the validation and validation report storage. Requires
+   * authentification to be set prior execution i.e. environment variable
+   * GOOGLE_APPLICATION_CREDENTIALS has to be defined.
+   */
+  public String run(
```

Comment:
> Is it possible to force Spring Boot to only use HTTPS without a lot of manual configuration? If not, you should also call out that HTTP communication is used and is inherently insecure. It would be good to default to HTTPS throughout, including in the documentation.

It is possible to require HTTPS, but from my early research it seems that this required to expose `.p12`. and `.crt` files that might contain credentials, on Github. I will look further, but I'd rather not publish certificates on GitHub - there might be a trick to avoid this.